### DISCUSSION TITLE
What's the difference between `$node` and `$nodeSchema`?

### DISCUSSION THREAD

**Discussion by ylc395** [Milkdown/milkdown#1152] [Q&A] [answered]
Which one should I choose?

**Comment by Saul-Mirone** [comment, marked as answer]
In most case you can use `$nodeSchema`.

`$nodeSchema` is a wrapper for `$node`, it'll provide more methods than `$node`, for example, `ctx` allows you to get the original schema spec and `extendSchema` gives you the ability to extend an existing schema.
